sql/memo/rel_props.go

```diff
@@ -327,14 +328,53 @@ func statsForRel(rel RelExpr) sql.Statistic {
 		// different joins use different ways to estimate cardinality of outputs
 		jp := rel.JoinPrivate()
 		left := jp.Left.RelProps.GetStats()
-		right := jp.Left.RelProps.GetStats()
+		right := jp.Right.RelProps.GetStats()
 
-		switch n := rel.(type) {
-		case *LookupJoin:
-			if n.Injective {
+		var injective bool
+		var mergeStats sql.Statistic
+		var n RelExpr = rel
+		var done bool
+		for n != nil && !done {
```

Comment:
`CardMemoGroups` is called once per join planning I think, so at most this should only traverse the group once. It could be shorter even, if we find an injective lookup there's no more info to check.